identity/user_test.go

```diff
@@ -0,0 +1,98 @@
+package identity_test
+
+import (
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/direct-state-transfer/dst-go/ethereum/test"
+	"github.com/direct-state-transfer/dst-go/identity"
+)
+
+func Test_Integ_NewUnlockedUser(t *testing.T) {
+	rng := rand.New(rand.NewSource(time.Now().UnixNano()))
+	setup, cleanup := test.NewUserSetup(t, rng, "../testdata")
+	defer cleanup()
+
+	type args struct {
+		wb  identity.WalletBackend
+		cfg identity.UserConfig
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    identity.User
+		wantErr bool
+	}{
+		{
+			name: "valid",
+			args: args{
+				wb: setup.WalletBackend,
+				cfg: identity.UserConfig{
+					Alias: setup.User.Alias,
+					OnChainWallet: identity.WalletConfig{
+						Addr:         setup.User.OnChainAcc.Address(),
+						KeystorePath: setup.KeystorePath,
+						Pwd:          ""},
+					OffChainWallet: identity.WalletConfig{
+						Addr:         setup.User.OnChainAcc.Address(),
+						KeystorePath: setup.KeystorePath,
+						Pwd:          ""}}},
+			wantErr: false,
+		},
+		{
+			name: "invalid-onchain-pwd",
+			args: args{
+				wb: setup.WalletBackend,
+				cfg: identity.UserConfig{
+					Alias: setup.User.Alias,
+					OnChainWallet: identity.WalletConfig{
+						Addr:         setup.User.OnChainAcc.Address(),
+						KeystorePath: setup.KeystorePath,
+						Pwd:          "invalid-pwd"}}},
+			wantErr: true,
+		},
+		{
+			name: "valid-onchain-invalid-offchain-pwd",
+			args: args{
+				wb: setup.WalletBackend,
+				cfg: identity.UserConfig{
+					Alias: setup.User.Alias,
+					OnChainWallet: identity.WalletConfig{
+						Addr:         setup.User.OnChainAcc.Address(),
+						KeystorePath: setup.KeystorePath,
+						Pwd:          ""},
+					OffChainWallet: identity.WalletConfig{
+						Addr:         setup.User.OnChainAcc.Address(),
+						KeystorePath: setup.KeystorePath,
+						Pwd:          "invalid-pwd"}}},
+			wantErr: true,
+		},
+		{
+			name: "invalid-keystore-path",
+			args: args{
+				wb: setup.WalletBackend,
+				cfg: identity.UserConfig{
+					Alias: setup.User.Alias,
+					OnChainWallet: identity.WalletConfig{
+						Addr:         setup.User.OnChainAcc.Address(),
+						KeystorePath: "invalid-keystore-path",
```

Comment:
Why does this work when i replace it with the correct `setup.KeystorePath`?

Reply:
Thanks for testing it out. It fails because `OffchainWallet` is not defined. Have fixed the test, not it should fail with correct keystore.